File: CodinGame/Easy/Mime Type/MimeType.cpp
#include <iostream>
#include <algorithm>
#include <string>
#include <unordered_map>

using namespace std;

int main()
{

    int N; // Number of elements which make up the association table.
    cin >> N;
    cin.ignore();
    int Q; // Number Q of file names to be analyzed.
    cin >> Q;
    cin.ignore();
    unordered_map<string, string> extenisonTypes;

    for (int i = 0; i < N; i++)
    {
        string EXT; // file extension
        string MT;  // MIME type.
        cin >> EXT >> MT;

        transform(begin(EXT), end(EXT), begin(EXT), ::tolower);

        extenisonTypes.insert(pair<string, string>(EXT, MT));

        cin.ignore();
    }
    for (int i = 0; i < Q; i++)
    {
        string FNAME;
        getline(cin, FNAME); // One file name per line.

        auto dot = find(rbegin(FNAME), rend(FNAME), '.');

        if (dot == rend(FNAME))
        {
            cout << "UNKNOWN" << endl;
            continue;
        }

        string extension;
        extension.resize(distance(rbegin(FNAME), dot));
        transform(rbegin(FNAME), dot, rbegin(extension), ::tolower);

        auto search = extenisonTypes.find(extension);

        if (search == extenisonTypes.end())
        {
            cout << "UNKNOWN" << endl;
            continue;
        }

        cout << search->second << endl;
    }
}

File: CodinGame/Easy/Mime Type/MimeType.cs
using System.Linq;
using System;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;

namespace MimeType
{
    static class Program
    {
        static void Main(string[] args)
        {
            int N = int.Parse(Console.ReadLine()); // Number of elements which make up the association table.
            int Q = int.Parse(Console.ReadLine()); // Number Q of file names to be analyzed.

            Dictionary<string, string> extensionTypes = new Dictionary<string, string>();

            for (int i = 0; i < N; i++)
            {
                string[] inputs = Console.ReadLine().Split(' ');
                string EXT = inputs[0]; // file extension
                string MT = inputs[1]; // MIME type.

                extensionTypes.Add(EXT.ToLower(), MT);
            }

            List<int> lista = new List<int> { 1, 2, 3 };

            for (int i = 0; i < Q; i++)
            {
                string FNAME = Console.ReadLine(); // One file name per line.

                if (FNAME.Contains('.'))
                {
                    string extension = FNAME.Substring(FNAME.LastIndexOf('.') + 1).ToLower();
                    if (extensionTypes.ContainsKey(extension))
                    {
                        Console.WriteLine(extensionTypes.First(w => w.Key == extension).Value);
                        continue;
                    }
                }

                Console.WriteLine("UNKNOWN");
            }

        }
    }
}